rfcs/2020-08-27-3603-postgres-metrics.md

```diff
@@ -0,0 +1,154 @@
+
+- `postgresql_up` -> Used as an uptime metric (0/1) ? - merits a broader discussion.
+- `pg_stat_database_blk_read_time` tagged with db, host, server, user (counter)
+- `pg_stat_database_blk_write_time` tagged with db, host, server, user (counter)
+- `pg_stat_database_blks_hit` tagged with db, host, server, user (counter)
+- `pg_stat_database_blks_read` tagged with db, host, server, user (counter)
+- `pg_stat_database_stats_reset` tagged with db, host, server, user(counter)
+- `pg_stat_bgwriter_buffers_alloc` tagged with db, host, server, user (counter)
+- `pg_stat_bgwriter_buffers_backend` tagged with db, host, server, user (counter)
+- `pg_stat_bgwriter_buffers_backend_fsync` tagged with db, host, server, user (counter)
+- `pg_stat_bgwriter_buffers_checkpoint` tagged with db, host, server, user (counter)
+- `pg_stat_bgwriter_buffers_clean` tagged with db, host, server, user (counter)
+- `pg_stat_bgwriter_checkpoint_sync_time` tagged with db, host, server, user (counter)
+- `pg_stat_bgwriter_checkpoint_write_time` tagged with db, host, server, user (counter)
+- `pg_stat_bgwriter_checkpoints_req` tagged with db, host, server, user (counter)
+- `pg_stat_bgwriter_checkpoints_time` tagged with db, host, server, user (counter)
+- `pg_stat_bgwriter_maxwritten_clean` tagged with db, host, server, user (counter)
+- `pg_stat_bgwriter_stats_reset` (counter)
+- `pg_stat_database_conflicts` tagged with db, host, server, user (counter)
+- `pg_stat_database_datid` tagged with db, host, server, user (counter)
+- `pg_stat_database_deadlocks` tagged with db, host, server, user (counter)
+- `pg_stat_database_numbackends` tagged with db, host, server, user (gauge)
+- `pg_stat_database_temp_bytes` tagged with db, host, server, user (counter)
+- `pg_stat_database_temp_files` tagged with db, host, server, user (counter)
+- `pg_stat_database_tup_deleted` tagged with db, host, server, user (counter)
+- `pg_stat_database_tup_fetched` tagged with db, host, server, user (counter)
+- `pg_stat_database_tup_inserted` tagged with db, host, server, user (counter)
+- `pg_stat_database_tup_returned` tagged with db, host, server, user (counter)
+- `pg_stat_database_tup_updated` tagged with db, host, server, user (counter)
+- `pg_stat_database_xact_commit` tagged with db, host, server, user (counter)
+- `pg_stat_database_xact_rollback` tagged with db, host, server, user (counter)
+- `postgresql_database_conflicts` tagged by db name and server (counter)
+- `postgresql_database_conflicts_confl_bufferpin` tagged by db name and server (counter)
+- `postgresql_database_conflicts_confl_deadlock` tagged by db name and server (counter)
+- `postgresql_database_conflicts_confl_lock tagged` by db name and server (counter)
+- `postgresql_database_conflicts_confl_snapshot` tagged by db name and server (counter)
+- `postgresql_database_conflicts_confl_tablespace` tagged by db name and server (counter)
+
+Naming of metrics is determined via:
+
+- `pg _ db_name _ column_name`
```

Comment:
I am unclear on what you're describing here. The metrics are named above as things like `pg_stat_database_temp_files` or `postgresql_database_conflicts_confl_bufferpin`, but here using a different pattern. Are the above column names which will be renamed to this pattern, or is the above a representation of this pattern? Does this imply that we will have high cardinality for the metric names, since they have database and column names embedded?

Reply:
Ah - I find/replaced some metric names and missed some. Will fix it up.

Reply:
I don't see that this question has been resolved in the recent changes. Perhaps an example would be helpful to illustrate, using one of the metrics listed above and showing the result with sample data.

Reply:
I see where it's misleading. I've re-written that section.